PacketTunnelProvider/NetworkProtection/NetworkProtectionPacketTunnelProvider.swift

```diff
@@ -469,20 +522,27 @@ final class NetworkProtectionPacketTunnelProvider: PacketTunnelProvider {
             wrappee: notificationsPresenter
         )
         notificationsPresenter.requestAuthorization()
+
+        let entitlementsCheck: (() async -> Result<Bool, Error>) = {
+            Logger.networkProtection.log("Subscription Entitlements check...")
+            let isNetworkProtectionEnabled = await subscriptionManager.isFeatureActive(.networkProtection)
+            Logger.networkProtection.log("NetworkProtectionEnabled if: \( isNetworkProtectionEnabled ? "Enabled" : "Disabled", privacy: .public)")
+            return .success(isNetworkProtectionEnabled)
```

Comment:
Not sure if valid, but earlier the closure returned `.failure` when fetching entitlement failed. Is this not possible for the new implementation of entitlements?